### ISSUE TITLE
SQL Server Input - String or Binary Data would be Truncated with WaitStats

### ISSUE THREAD

**Issue by jgerschk** [influxdata/telegraf#1854]
## Bug report

[Include Telegraf version, operating system name, and other relevant details]

1.0.1 - Telegraf

I think there is a typo here within the SQL code.

```
DECLARE @w4 TABLE
(
	WaitType nvarchar(64) NOT NULL,
	WaitCategory nvarchar(64) NOT NULL
)
DECLARE @w5 TABLE
(
	WaitCategory nvarchar(16) NOT NULL,
	WaitTimeInMs bigint NOT NULL,
	WaitTaskCount bigint NOT NULL
)
```

Within the @w5 declaration of table, should not WaitCategory be 64 as well, and not 16.

After installing the 1.0.1 version (can't say if it worked before since this is new), we see errors in the log of String or binary data would be truncated, and not wait state data is being sent.

So I suspect that this line needs to change to nvarchar(64)

**Comment by sparrc** [comment]
cc @zensqlmonitor what do you think? does nvarchar(16) need to be changed to nvarchar(64)?

**Comment by zensqlmonitor** [comment]
yes, typo error.
You are the first who meets the DISTRIBUTED QUERY category (with a length of 17 !)
I will modify the code and the new PR.

**Comment by w4** [comment]
I got tagged here 😢

**Comment by sparrc** [comment]
LOL, sorry @w4

**Comment by jgerschk** [comment]
Thanks @sparrc and @zensqlmonitor !  Well, at least @w4 you can hold more! :)

**Comment by sparrc** [comment]
closed by  #1859
